sdk/dotnet/Pulumi.Automation/ProjectSettings.cs

```diff
@@ -44,5 +47,57 @@ public class ProjectSettings
 
         internal static ProjectSettings Default(string name)
             => new ProjectSettings(name, new ProjectRuntime(ProjectRuntimeName.NodeJS));
+
+        internal bool IsDefault
+        {
+            get
+            {
+                return ProjectSettings.Comparer.Equals(this, ProjectSettings.Default(this.Name));
+            }
+        }
+
+        internal static IEqualityComparer<ProjectSettings> Comparer { get; } = new ProjectSettingsComparer();
+
+        private sealed class ProjectSettingsComparer : IEqualityComparer<ProjectSettings>
+        {
+            bool IEqualityComparer<ProjectSettings>.Equals(ProjectSettings? x, ProjectSettings? y)
+            {
+                if (x == null)
+                {
+                    return y == null;
+                }
+
+                if (y == null)
+                {
+                    return x == null;
+                }
+
+                return x.Name == y.Name &&
```

Comment:
I'm wondering if for string comparison we may want to use `string.Equals(...)` and decide on a `StringComparison` enum value that works for our use-case so we support culture-specific string comparison

Reply:
I'd pass on that for the moment since it's hard to know how deep this should go.

Reply:
Sounds good. Something to keep in mind, just because we're comparing strings from a consumer-controlled file so it might not always be US english or whatever